src/code/examples/movable-camera.rs

```diff
@@ -0,0 +1,373 @@
+fn adjust_cam_speed<P>(
+   time: Res<Time>,
+   keys: Res<Input<KeyCode>>,
+   mut cam: Query<&mut MovableCamera>,
+) where
+   P: MovableCameraParams,
+{
+   let mut cam = cam.single_mut();
+   if keys.just_pressed(P::CHANGE_SPEED) {
+      cam.slow = !cam.slow;
+      if !cam.slow {
+         cam.speed = P::DEFAULT_SPEED;
+      }
+   }
+
+   if cam.slow {
+      cam.speed = P::SLOW_SPEED;
+      cam.angular_speed = P::SLOW_SPEED;
+   } else if keys.any_pressed([
+      P::LEFT,
+      P::RIGHT,
+      P::UPWARD,
+      P::BACKWARD,
+      P::DOWNWARD,
+      P::FORWARD,
+   ]) {
+      cam.speed += CameraParams::ACCELERATION.mul(time.delta_seconds());
+   } else {
+      cam.speed = CameraParams::DEFAULT_SPEED;
+      cam.angular_speed = CameraParams::DEFAULT_SPEED;
+   }
+}
+
+/// Move the camera with QWEASD, zoom with wheel, focus at
+/// camera pos with F, and rotate/orbit with right mouse button.
+fn movable_camera<P>(
+   windows: ResMut<Windows>,
+   time: Res<Time>,
+   keys: Res<Input<KeyCode>>,
+   buttons: Res<Input<MouseButton>>,
+   mut motion: EventReader<MouseMotion>,
+   mut scroll_evr: EventReader<MouseWheel>,
+   mut q_child: Query<(
+      &Parent,
+      &mut Transform,
+      &mut MovableCamera,
+      &PerspectiveProjection,
+   )>,
+   mut q_parent: Query<(&mut Transform, &GlobalTransform), Without<PerspectiveProjection>>,
+) where
+   P: MovableCameraParams,
+{
+   for (parent, mut transform_child, mut cam, ..) in q_child.iter_mut() {
+      // Focused Camera
+      if cam.focused {
+         if keys.any_pressed([
+            CameraParams::LEFT,
+            CameraParams::RIGHT,
+            CameraParams::UPWARD,
+            CameraParams::BACKWARD,
+            CameraParams::DOWNWARD,
+            CameraParams::FORWARD,
+         ]) {
+            if let Ok((mut transform_parent, ..)) = q_parent.get_mut(parent.0) {
+               let zoom = transform_child.translation.z;
+               // Set child transform to parent transform
+               *transform_child = *transform_parent;
+               // Offset child by its zoom
+               transform_child.translation += zoom.mul(transform_parent.back());
+               // Set parent transform to origin
+               *transform_parent = Transform::default();
+            }
+            cam.focused = false;
+         }
+      } else if keys.just_pressed(P::FOCUS) {
+         if let Ok((mut transform_parent, ..)) = q_parent.get_mut(parent.0) {
+            // Hand off position and orientation information to parent
+            *transform_parent = *transform_child;
+         }
+         *transform_child = Transform::default();
+         cam.focused = true;
+      }
+
+      let mut rotation_move = Vec2::ZERO;
+      let mut scroll = 0.0;
+
+      if buttons.pressed(MouseButton::Right) {
+         for ev in motion.iter() {
+            rotation_move += ev.delta;
+         }
+      }
+
+      for ev in scroll_evr.iter() {
```

Comment:
```suggestion
      for event in scroll_events.iter() {
```

Characters are cheap ;)